### ISSUE TITLE
Speed

### ISSUE THREAD

**Issue by erikstr** [organicinternet/magento-configurable-simple#27]
Hi Matt
I read about the speed issues of the extension but did not believe that the impact was so heavy until we imported a couple of config products with 1200 - 4300 associated simple products. Now the page load times range from 10 - 15 sec. which is absolutely not acceptable. We are hosted on an i7 dedicated server with 8GB of RAM, so it is no solution targeting a hardware upgrade. 

However I saw your post here:

http://www.magentocommerce.com/boards/viewthread/43134/P15/#t285198

that a faster parsing method could be used. As we are on Magento 1.5, I would like to disable the slow method at all and use only the fast one. Could you tell me which code to alter in order to achieve this in a clean manner? Or is the quick fix from the guy there the right solution?

I would be also glad to know how to disable some Magento standard checks for which I might be sure that they are not needed in order to further increase the speed of the installation. 

Thanks in advance,
Erik

**Comment by nerotic** [comment]
I too would be interested in the very same thing.  We only have 44 simple products comprising our configurable products and we're at 8 seconds in page load.

I had read that same thread but the code to be altered isn't there in 0.8.0.

Thanks.

**Comment by erikstr** [comment]
I am still hoping that Matt or some other smart guy will post a clean solution to 0.8.0. We have two weeks before we have to solve the issue by looking into the code ourselves. While I am sure that we will manage this, it will be of no use to others if Matt doesn't take the changes into his official releases. SCP is hardly usable with such loading times and I can not imagine that someone is running it on a live shop.

**Comment by nerotic** [comment]
Any word on this at all? 

Thanks.

**Comment by erikstr** [comment]
Here is my update on this issue:
We started modifying the SCP code and achieved good results by speeding up SCP itself. Load time dropped from 70 (!) seconds down to 32-34 sec (wow, what a blazing speed). At this point (stupid us, we could have used the profiler first) we realised that not only SCP is that slow but Magento itself. We have approx. 24 product attributes, most of them drop-down-type. It takes ages to get and show all these attributes too. 

Next try: we abandoned the idea of code optimisation and looked into the mostly not used and per default deactivated Magento cache. The Magento cache has to be enabled in the constructor function of each and every block class - with all of its exceptions and tweaks. After a couple of hours the answer was: "No, thanks!"

So now we are heading to the only logical and hopefully usable solution: full page caching with Varnish. We haven't used Varnish before so we'll have to learn first. There is an extension for €300 which is worth the money but it is completely closed (encrypted PHP code) and I don't have a nice feeling when I don't know what is going on and how. We would however buy it if we can't manage to do it ourselves using these tutorials and the free module:
http://www.kalenyuk.com.ua/magento-performance-optimization-with-varnish-cache-47.html
http://moprea.ro/2011/feb/16/magento-performance-optimization-varnish-cache-2/
http://moprea.ro/2011/may/6/magento-performance-optimization-varnish-cache-3/

I am however not closing the issue because SCP would become more usable if _it's_ speed problems gets resolved.

**Comment by nerotic** [comment]
Hey Erik,

Thanks for the information.  Since we're on shared hosting we're taking a different route.  

I did some research and saw this: http://magebenchmark.sonassi.com/hosts/us/   Sonassi ranked DX3 even higher than their own Magento specific hosting so I went and did some information gathering. 

DX3 offers a one month trial of their service so I contacted them on Sunday. Since then I've been dealing with Andy who is my account manager.

So far the customer service has been top shelf. They've been extremely responsive and just yesterday they activated my free test account. They've listened to all my concerns and have been straightforward and honest in their dealings so far. 

Plus for 15 pounds a months it doesn't really involve any additional costs per month if we end up switching since we're already spending that. 

It's also managed hosting with their team of Magento experts dealing with the hard core issues.

I know this sounds like an ad (it isn't) but you can find more info here: http://dx3webs.com/front/

I'll be porting our exact setup onto their servers for testing this weekend, I'll post back with my initial observations and conclusions.

**Comment by nerotic** [comment]
Hi Erik,

Well I guess it's just the two of us :)

Anyway, I got one of our Magento installs up and running on the Dx3's test server and the results are amazing:

http://nerotic.net/auxout/levelown.php

Only the USA store is pointing to our the test server.  If you choose EU or EU VAT in the upper right hand corner you'll get the stores on our current host.

The speed gains here are remarkable, shaved off at least 66% of our load time just in changing host. This is out of the box, they have a series of optimizations that they're going to run today that should improve the speed quite a bit more.

You can try it for yourself.

**Comment by erikstr** [comment]
Yes, it seems that nobody else suffers from there speed issues ...

Regarding your site - you've maybe already moved all three sites to the US host, as for me they load equally fast - for approx. 3 sec. (I am in Switzerland).

These are also the times we managed to achieve with our configurables before the actual data delivery from Apache by block caching the Product View page. However, our "bulkiest" product has 1444 different options and this, combined with the 37 attributes, generates a huge javascript, which has to be downloaded to the browser. So we need another 3 sec. for the download alone!

A very useful feature for SCP would be, if the sub-products' infos were being pulled with ajax instead of loading all of them on first run.

memcached und apc didn't bring more speed, at least without heavy server load, but we left memcached running, just in case ... We dropped Varnish for the moment, as we have a lot of other work to do.

nerotic, why have you set up two different EU stores? What is your issue with the VAT and how are you going to invoice the different VATs for different EU countries. I know that there is an extension for this, but you seem to have gone a different way ... We are also facing this problem and have chosen to make a separate website for each country.

**Comment by madpuppy** [comment]
also having this speed problem... with extension switched on generating the product html takes 30-60 seconds, without extension less than 6.

**Comment by e-citron** [comment]
Hi guys,

Did you find any solution to speed up SCP when having a big Super Attribute? We achieve to have a 6sec to 10 sec page load, but it's still really slow ! How did you manage to solve the issue?

Thanks !

**Comment by jfrubio** [comment]
Hi.

We have same trouble, we have configurable products with 4000 simple products and load time takes more than 60 sec.

We have a dedicated server 2x6cores Xeon 3.500mhz, 2x500gb SSD raid disk, 64gb ram 1800mhz. Windows 2012 with wamp. MySQL 5.6.21 and Magento 1.7

We need to lower load time for magento configurable products.
